### DISCUSSION TITLE
Often get 503 error

### DISCUSSION THREAD

**Discussion by long-nguyen** [weather-gov/api#608] [Q&A] [answered]
We often get 503 error when calling grid API, does anyone know how to fix?
```
{
    "correlationId": "8ddee8dc",
    "title": "Service Unavailable",
    "type": "https://api.weather.gov/problems/ServiceUnavailable",
    "status": 503,
    "detail": "An upstream data source is temporarily unavailable. Please try again later.",
    "instance": "https://api.weather.gov/requests/8ddee8dc"
}
```

**Comment by justinlangley3** [comment, marked as answer]
If you're using the `/gridpoints/ ` route , it appears to be having issues over the holiday. I'm seeing http 503 status, as well.
This GitHub repo is not for reporting outages or service issues.

NOAA requests for these issues to be reported to the email listed on this webpage:
https://weather-gov.github.io/api/reporting-issues

**Reply by JJKraw** [reply]
And there have been plenty of reports (I'm on the email thread...)

**Reply by scadergit** [reply]
This is due to a service failure upstream from the API. It has been on the schedule to replace for some time, but always behind other priorities. Hopefully, we can address this issue before long now that our development team has increased.

**Reply by long-nguyen** [reply]
Thanks, hope it could be resolved soon

**Comment by KCRotaryUSA** [comment]
I get a 503 for every location. Have sent email to nco.ops@noaa.gov

**Reply by isagonza** [reply]
Today I was looking at the data that I collect everyday and only collected 2 out of 112 places :( I hope the API starts working again soon.

**Comment by grantiago** [comment]
I got a nice reply by emailing nco.ops@noaa.gov In a nutshell -- blame-passing to upstream sources. Nonetheless the 503 errors persist. The API is worthless. Passing the buck. Classic.  Have a donut. Hold out for your pension at 65.

**Reply by otherjoel** [reply]
Did you not read the explanation above? I’ve been using this API happily for several years, it is definitely not worthless. If you need a mission-critical weather API, get out your wallet and go to weatherstack or similar.

**Reply by grantiago** [reply]
I see no explanation above. I see excuses, passing-the-buck, not taking responsibility. I too have been using this API for years.

**Comment by akrherz** [comment]
Latest [NCO ADM](https://www.nco.ncep.noaa.gov/status/messages/)

```
SENIOR DUTY METEOROLOGIST NWS ADMINISTRATIVE MESSAGE
NWS NCEP CENTRAL OPERATIONS COLLEGE PARK MD
0112Z FRI DEC 30 2022


...ERRORS WITH API.WEATHER.GOV GRIDPOINT FORECASTS - UPDATE...

NCO support teams worked to stabilize the API gridpoints forecast
service by increasing the amount of resources needed for the
service to function properly...users should now be able to access
the API gridpoints forecast service.  Root cause for this issue
was believed to be due to higher than normal user traffic during
last week's winter storm event, that correlated to when the
service became degraded and couldn't fully recover.  If users
continue to experience issues, please contact Tech Control at
nco.ops@noaa.gov or by phone at (301) 683-1518.
```

**Comment by scadergit** [comment]
The upstream service translates the netcdf to json, so the API did not have data to satisfy requests. No bucks, just how it works.

NCO has dedicated additional resources to accelerate the modernization effort, but we also have to balance keeping the current systems alive. 

I understand the frustration, and you've all been very patient. We'll continue to stabilize the services and finally have the opportunity to focus on feature enhancement.

**Reply by JJKraw** [reply]
Truly appreciate the effort to right the ship.
